container/pull.bzl

```diff
@@ -157,9 +157,11 @@ exports_files(["image.digest", "digest"])
 
     kwargs = {}
 
-    # TODO(suvanjan): add the PULLER_TIMEOUT as a environment variable for the puller to process.
-    # if "PULLER_TIMEOUT" in repository_ctx.os.environ:
-    #     kwargs["timeout"] = int(repository_ctx.os.environ.get("PULLER_TIMEOUT"))
+    if "PULLER_TIMEOUT" in repository_ctx.os.environ:
+        args += [
+            "--timeout",
+            repository_ctx.os.environ.get("PULLER_TIMEOUT"),
+        ]
 
     result = repository_ctx.execute(args, **kwargs)
```

Comment:
`timeout` needs to be in kwargs as well, otherwise it will default to `600`. [bazel docs link](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#parameters-3). 

At the moment the puller executable is respecting the timeout but bazel is enforcing a hard limit of 600 seconds.